### DISCUSSION TITLE
Need help configuring Authelia with PostgreSQL DB

### DISCUSSION THREAD

**Discussion by slax81** [authelia/authelia#4429] [Q&A] [answered]
Hello. I'm trying to move Authelia in Kubernetes from SQLite to Percona PG operator. Other apps are working ok, so it must be something misconfigured in Authelia. I get this error on start:

`level=error msg="Failure running the storage provider startup check: error pinging database: failed to connect to `host=pg-db-pgbouncer.percona.svc.cluster.local user=authelia database=authelia`: server error (FATAL: unsupported startup parameter: search_path (SQLSTATE 08P01))"`

Please let me know if you have any ideas on how I could overcome this problem. Thanks

**Comment by slax81** [comment]
I forgot to mention, when I change host url directly to DB instead pgbouncer I get auth error (user with password exists and also database)
`time="2022-11-24T13:00:19Z" level=info msg="Authelia v4.37.0 is starting"
time="2022-11-24T13:00:19Z" level=info msg="Log severity set to info"
time="2022-11-24T13:00:29Z" level=error msg="Failure running the storage provider startup check: error pinging database: failed to connect to `host=pg-db.percona.svc.cluster.local user=authelia database=authelia`: failed SASL auth (FATAL: password authentication failed for user \"authelia\" (SQLSTATE 28P01))" stack="github.com/authelia/authelia/v4/internal/commands/root.go:277 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.0/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.0/command.go:1040                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.0/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"
time="2022-11-24T13:00:29Z" level=warning msg="Notifier SMTP connection has opportunistic STARTTLS explicitly disabled which means all emails will be sent insecurely over plain text and this setting is only necessary for non-compliant SMTP servers which advertise they support STARTTLS when they actually don't support STARTTLS"
time="2022-11-24T13:00:29Z" level=fatal msg="The following providers had fatal failures during startup: storage" stack="github.com/authelia/authelia/v4/internal/commands/root.go:305 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.0/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.0/command.go:1040                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.0/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"`

**Comment by james-d-elliott** [comment, marked as answer]
pgbauncer has issues with the search_path parameter since it's not actually postgres I don't believe, just it imitates the protocol, you can tell pgbauncer to ignore this though with the pgbauncer config `IGNORE_STARTUP_PARAMETERS: search_path` [see here](https://www.appsloveworld.com/postgresql/100/222/postgressql-org-postgresql-util-psqlexception-error-unsupported-startup-parame).

Your second error is a username/password error with percona.
